### PR TITLE
feat(es/lint): quotes rule implementation

### PR DESCRIPTION
**Pull request by ArturAralin**
**Description:**

"quotes" rule implementation

**BREAKING CHANGE:**
No

### CODE REVIEW

File: crates/swc/tests/errors/lints/quotes/valid-cases-backtick/input.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+() => {
+    "use strict";
+    "use strong";
+    "use asm";
+    var foo = `backtick`;
+};
+
+// TODO: "error: 'import', and 'export' cannot be used outside of module code"
```
Comment (ArturAralin) [inline crates/swc/tests/errors/lints/quotes/valid-cases-backtick/input.js:24]: @kdy1 why tests throws me this error? I was dive in code of test and it marks code as module

Comment (kdy1) [reply crates/swc/tests/errors/lints/quotes/valid-cases-backtick/input.js:24]: I don't know

Comment (kdy1) [reply crates/swc/tests/errors/lints/quotes/valid-cases-backtick/input.js:24]: Maybe `IsModule` is used?

Comment (ArturAralin) [reply crates/swc/tests/errors/lints/quotes/valid-cases-backtick/input.js:24]: Yes, It used

File: crates/swc_ecma_lints/src/rules/quotes.rs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+#[derive(Default)]
+struct Quotes {
+    expected_reaction: LintRuleReaction,
+    prefer: QuotesType,
+    avoid_escape: bool,
+    allow_template_literals: bool,
+    source_map: Arc<SourceMap>,
+}
+
+impl Debug for Quotes {
```
Comment (ArturAralin) [inline crates/swc_ecma_lints/src/rules/quotes.rs:60]: Custom Debug implemented cause SourceMap haven't Debug impl

**Review by kdy1 — APPROVED** [review]
Thanks!

**Pull request merged**